data/src/main/java/com/microsoft/azure/kusto/data/BaseClient.java

```diff
@@ -142,8 +153,13 @@ public static DataServiceException createExceptionFromResponse(String url, HttpR
                 isPermanent);
     }
 
+    private Context getContextTimeout(long timeoutMs) {
+        int requestTimeout = timeoutMs > Integer.MAX_VALUE ? Integer.MAX_VALUE : Math.toIntExact(timeoutMs) + CLIENT_SERVER_DELTA_IN_MILLISECS;
```

Comment:
Maybe we should have a lower hard cap than MAX_VALUE, I think in the server side it's 1 hour